src/mixtures/mixturemodel.jl

```diff
@@ -125,7 +125,7 @@ the components given by ``params``, and a prior probability vector.
 If no `prior` is provided then all components will have the same prior probabilities.
 """
 function MixtureModel(::Type{C}, params::AbstractArray) where C<:Distribution
-    components = C[_construct_component(C, a) for a in params]
+    components = [_construct_component(C, a) for a in params]
```

Comment:
Why was this changed?

Reply:
Because `C` is often not used in a parametrised way (e.g. `Normal`, since the eltype can be inferred from `params`), using `C[]` resulted in the code generating a vector of an abstract type. Just creating the vector creates a concrete type though, as all elements are the same.